lms/djangoapps/grades/signals/handlers.py

```diff
@@ -244,6 +247,13 @@ def recalculate_course_grade(sender, course, course_structure, user, **kwargs):
     CourseGradeFactory().update(user, course=course, course_structure=course_structure)
 
 
+@receiver(ENROLLMENT_TRACK_UPDATED)
+@receiver(COHORT_MEMBERSHIP_UPDATED)
+def recalculate_course_grade_on_user_partition_change(sender, user, course_key, **kwargs):
```

Comment:
Why create a new method here?  Why not add the additional receivers to the existing `recalculate_course_grade` method above?
You can add the `course_key` parameter as an optional parameter to the method above.  The `CourseGradeFactory` is then smart to determine which of the course objects/identifiers to use (via the `CourseData` object.

Reply:
The distinction between this and the above method is that this one passes `force_update_subsections=True` to the course grade factory. Having two short methods seemed simpler than trying to have everything in one method that tries to distinguish between different signals in order to determine whether to pass that kwarg.